pkg/apis/core/validation/shoot_test.go

```diff
@@ -2129,6 +2129,73 @@ var _ = Describe("Shoot Validation Tests", func() {
 
 			Expect(errorList).To(HaveLen(0))
 		})
+
+		Describe("kubeconfig rotation", func() {
+			DescribeTable("DisallowKubeconfigRotationForShootInDeletion=true",
+				func(oldAnnotations, newAnnotations map[string]string, newSetDeletionTimestamp, expectedError bool) {
+					defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.DisallowKubeconfigRotationForShootInDeletion, true)()
+					now := metav1.NewTime(time.Now())
+					newShoot := prepareShootForUpdate(shoot)
+					if oldAnnotations != nil {
+						shoot.Annotations = oldAnnotations
+					}
+
+					if newSetDeletionTimestamp {
+						newShoot.DeletionTimestamp = &now
+					}
+					newShoot.Annotations = newAnnotations
+
+					errorList := ValidateShootObjectMetaUpdate(newShoot.ObjectMeta, shoot.ObjectMeta, field.NewPath("metadata"))
+
+					if expectedError {
+						Expect(errorList).ToNot(HaveLen(0))
+						Expect(errorList).To(ConsistOfFields(Fields{
+							"Type":   Equal(field.ErrorTypeInvalid),
+							"Field":  Equal("metadata.annotations[gardener.cloud/operation]"),
+							"Detail": ContainSubstring(`kubeconfig rotations is not allowed for clusters in deletion`),
+						}))
+					} else {
+						Expect(errorList).To(HaveLen(0))
+					}
+				},
+				Entry("it should allow kubeconfig rotation for cluster not in deletion", nil, map[string]string{"gardener.cloud/operation": "rotate-kubeconfig-credentials"}, false, false),
```

Comment:
Nit: the idiomatic msg is without `it`

```suggestion
				Entry("should allow kubeconfig rotation for cluster not in deletion", nil, map[string]string{"gardener.cloud/operation": "rotate-kubeconfig-credentials"}, false, false),
```